src/VisualStudio/Core/Def/Implementation/FindReferences/StreamingFindUsagesPresenter.cs

```diff
@@ -137,38 +137,28 @@ public void ClearAll()
         /// <param name="title"></param>
         /// <param name="supportsReferences"></param>
         /// <returns></returns>
-        public FindUsagesContext StartSearch(string title, bool supportsReferences, CancellationToken cancellationToken)
-        {
-            this.AssertIsForeground();
-            var context = StartSearchWorker(title, supportsReferences, includeContainingTypeAndMemberColumns: false, includeKindColumn: false, cancellationToken);
-
-            // Keep track of this context object as long as it is being displayed in the UI.
-            // That way we can Clear it out if requested by a client.  When the context is
-            // no longer being displayed, VS will dispose it and it will remove itself from
-            // this set.
-            _currentContexts.Add(context);
-            return context;
-        }
+        public (FindUsagesContext context, CancellationToken cancellationToken) StartSearch(string title, bool supportsReferences)
+            => StartSearchWithCustomColumns(title, supportsReferences, includeContainingTypeAndMemberColumns: false, includeKindColumn: false);
 
         /// <summary>
         /// Start a search that may include Containing Type, Containing Member, or Kind information about the reference
         /// </summary>
-        public FindUsagesContext StartSearchWithCustomColumns(
-            string title, bool supportsReferences, bool includeContainingTypeAndMemberColumns, bool includeKindColumn, CancellationToken cancellationToken)
+        public (FindUsagesContext context, CancellationToken cancellationToken) StartSearchWithCustomColumns(
+            string title, bool supportsReferences, bool includeContainingTypeAndMemberColumns, bool includeKindColumn)
         {
             this.AssertIsForeground();
-            var context = StartSearchWorker(title, supportsReferences, includeContainingTypeAndMemberColumns, includeKindColumn, cancellationToken);
+            var context = StartSearchWorker(title, supportsReferences, includeContainingTypeAndMemberColumns, includeKindColumn);
 
             // Keep track of this context object as long as it is being displayed in the UI.
             // That way we can Clear it out if requested by a client.  When the context is
             // no longer being displayed, VS will dispose it and it will remove itself from
             // this set.
             _currentContexts.Add(context);
-            return context;
+            return (context, context.CancellationTokenSource!.Token);
```

Comment:
See comment earlier, I'm not sure why CancellationTokenSource is nullable.

Reply:
In this case, I think it's too make sure we don't accidentally double dispose. If it's safe to do that, then we can remove the null out